File: commonfamily/src/main/java/com/koenig/StringFormats.java
package com.koenig;

import com.koenig.commonModel.User;

import org.joda.time.DateTime;

import java.text.DecimalFormat;
import java.text.DecimalFormatSymbols;
import java.util.List;
import java.util.Locale;

/**
 * Created by Thomas on 23.10.2017.
 */

public class StringFormats {

    public static final String dayFormat = "dd.MM.yy";
    private static final DecimalFormatSymbols symbols = buildSymbols();
    private static DecimalFormat centFormat = new DecimalFormat("0.00", symbols);
    private static DecimalFormat euroFormat = new DecimalFormat("0", symbols);
    private static DecimalFormat percentFormat = new DecimalFormat("0.#", symbols);

    private static DecimalFormatSymbols buildSymbols() {
        DecimalFormatSymbols symbols = new DecimalFormatSymbols(Locale.getDefault());
        symbols.setDecimalSeparator('.');
        symbols.setGroupingSeparator(',');
        return symbols;
    }

    public static String dateTimeToDayString(DateTime dateTime) {
        return dateTime.toString(dayFormat);
    }

    public static String centsToCentString(int cents) {
        return StringFormats.centFormat.format(cents / 100.0);
    }

    public static String centsToEuroString(int cents) {
        return StringFormats.euroFormat.format(cents / 100.0);
    }

    public static String floatToPercentString(float percent) {
        return StringFormats.percentFormat.format(percent * 100.0);
    }


    public static String usersToAbbreviationString(List<User> users) {
        if (users.size() == 0) return "";
        StringBuilder builder = new StringBuilder();
        for (User user : users) {
            builder.append(user.getAbbreviation() + ",");
        }

        return builder.substring(0, builder.length() - 1);
    }
}

File: commonfamily/src/main/java/com/koenig/StringFormats.kt
package com.koenig

import com.koenig.commonModel.User
import org.joda.time.LocalDate
import java.text.DecimalFormat
import java.text.DecimalFormatSymbols
import java.util.*

/**
 * Created by Thomas on 23.10.2017.
 */

object StringFormats {

    val dayFormat = "dd.MM.yy"
    private val symbols = buildSymbols()
    private val centFormat = DecimalFormat("0.00", symbols)
    private val euroFormat = DecimalFormat("0", symbols)
    private val percentFormat = DecimalFormat("0.#", symbols)

    private fun buildSymbols(): DecimalFormatSymbols {
        val symbols = DecimalFormatSymbols(Locale.getDefault())
        symbols.decimalSeparator = '.'
        symbols.groupingSeparator = ','
        return symbols
    }

    fun localDateTimeToDayString(day: LocalDate): String {
        return day.toString(dayFormat)
    }

    fun centsToCentString(cents: Int): String {
        return StringFormats.centFormat.format(cents / 100.0)
    }

    fun centsToEuroString(cents: Int): String {
        return StringFormats.euroFormat.format(cents / 100.0)
    }

    fun floatToPercentString(percent: Float): String {
        return StringFormats.percentFormat.format(percent * 100.0)
    }


    fun usersToAbbreviationString(users: List<User>): String {
        if (users.size == 0) return ""
        val builder = StringBuilder()
        for (user in users) {
            builder.append(user.abbreviation + ",")
        }

        return builder.substring(0, builder.length - 1)
    }
}
